### PR TITLE
Add Suffix Array Implementation using Manber-Myers Algorithm

### PR DESCRIPTION
**Pull request by mohit07raghav19**
- Implemented the **Manber-Myers algorithm** for constructing _suffix arrays_.
- Added comprehensive test cases to validate the correctness of the implementation.

# Pull Request Template

## Description

The implementation of a suffix array construction
algorithm based on the Manber-Myers algorithm(_optimized_ algorithm used for constructing suffix arrays). The `suffix_array_manber_myers`
function efficiently generates a suffix array for a given input string. The
implementation has been tested and validated with various test
cases, ensuring correct and efficient suffix array construction.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I ran bellow commands using the latest version of **rust nightly**.
- [x] I ran `cargo clippy --all -- -D warnings` just before my last commit and fixed any issue that was found.
- [x] I ran `cargo fmt` just before my last commit.
- [x] I ran `cargo test` just before my last commit and all tests passed.
- [x] I checked `COUNTRIBUTING.md` and my code follows its guidelines.

This closes a task of Suffix Array under Strings in #344.

I would request if you can add label:`hactoberfest-accepted` to this PR if applicable. :)

### CODE REVIEW

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by mohit07raghav19** [comment]
Thank you @siriak , 

I wanted to kindly request that the `hacktoberfest-accepted` label be added to my recently merged PR (https://github.com/TheAlgorithms/Rust/pull/529). 

I believe this PR aligns with the criteria and guidelines for Hacktoberfest, as it represents a meaningful contribution to the project. However, I understand that the final decision rests with you, the maintainers, and I respect your judgment.

**Comment by siriak** [comment]
I didn't have to add the label for the contribution to count because the repo is tagged with `hacktoberfest`. At least it used to be this way last year. But I've added it anyway by your request.

**Pull request merged**
